src/Integration.Vsix.UnitTests/CFamily/ProcessRunnerTests.cs

```diff
@@ -409,7 +409,90 @@ public void ProcRunner_DoNotLogSensitiveData()
             DummyExeHelper.AssertExpectedLogContents(exeLogFile, allArgs);
         }
 
-        #endregion Tests
+        [TestMethod]
+        public void Execute_CancellationTokenIsAlreadyCancelled_ProcessNotExecuted()
+        {
+            var exeName = WriteBatchFileForTest(TestContext,
+                @"@echo Hello world
+xxx yyy
+@echo Testing 1,2,3...>&2
+");
+
+            var logger = new TestLogger();
+            var args = new ProcessRunnerArguments(exeName, true);
+            var runner = CreateProcessRunner(logger);
+
+            args.CancellationToken = new CancellationToken(true);
+
+            // Act
+            var success = runner.Execute(args);
+
+            // Assert
+            success.Should().BeFalse("");
+            runner.ExitCode.Should().Be(0, "Unexpected exit code");
+
+            logger.AssertNoOutputMessages(); 
+        }
+
+        [TestMethod]
+        public async Task Execute_CancellationTokenCancelledMidway_ProcessKilled()
+        {
+            var exeName = WriteBatchFileForTest(TestContext,
+$@"
+@echo Hello world
+waitfor /t 4 { Guid.NewGuid():N}
+@echo Done!
+");
+
+            var logger = new TestLogger();
+            var cancellationTokenSource = new CancellationTokenSource();
+            var args = new ProcessRunnerArguments(exeName, true)
+            {
+                TimeoutInMilliseconds = 4000,
+                CancellationToken = cancellationTokenSource.Token
+            };
+
+            var runner = CreateProcessRunner(logger);
+
+            bool? result = null;
+
+            var runProcess = Task.Run(() =>
+            {
+                result = runner.Execute(args);
+            });
+
+            cancellationTokenSource.CancelAfter(100);
+
+            await runProcess;
+
+            result.Should().BeFalse("Expecting the process to have failed");
+            runner.ExitCode.Should().Be(-1, "Unexpected exit code");
+            logger.AssertOutputStringExists("Hello world");
```

Comment:
This looks like a potentially flaky check; it relies on the process having gotten far enough to write the first line before being cancelled.
It would be worth adding a comment calling it out as potentially flaky/timing dependent.